Add the option to include an extension in a phone number

jimmynotjim opened the pull request:
We have a need to include phone extensions when adding a number to
contact details but the current field does not allow for the extra
characters. To avoid differences in formatting, we've added a new field
for the extension digits that are then automatically formatted on the
front end.

## Additions

- New `extension` option to the wagtail contact phone molecule
- New `ext` formatting on the front-end of the contact phone molecule

## Testing

1. `tox -e fast v1.tests.atomic_elements.organisms.test_organisms`
2. Build a sub-landing page and include an expandable.
3. Add contact phone molecule and include a phone number and extension.
4. Verify the extension is formatted on the page.

## Screenshots

![screen shot 2017-11-07 at 1 17 17 pm](https://user-images.githubusercontent.com/1280430/32512860-00b12ae0-c3be-11e7-8997-ddeb8bd4f66f.png)

![screen shot 2017-11-07 at 1 17 14 pm](https://user-images.githubusercontent.com/1280430/32512862-01fd38c6-c3be-11e7-9116-ea4d67cd2921.png)

## Checklist

* [x] PR has an informative and human-readable title
* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [x] Passes all existing automated tests
* [x] Any *change* in functionality is tested
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated
* [x] Reviewers requested with the [Reviewer tool](https://help.github.com/articles/about-pull-request-reviews/) :arrow_right:

sarahsimpson09 commented:
@jimmynotjim this looks great, thanks! One thing I'm curious about--will the extension show up next to all versions of the phone number (vanity, tty) when entered? Or just once with the main phone?

jimmynotjim commented:
> will the extension show up next to all versions of the phone number (vanity, tty) when entered

@sarahsimpson09 Oh, good point, I didn't think to add it to those. Is it safe to assume the TTY extension might not match the primary extension and would need it's own field?

Scotchester commented on cfgov/v1/atomic_elements/molecules.py:
```diff
@@ -257,6 +257,7 @@ class ContactPhone(blocks.StructBlock):
         blocks.StructBlock([
             ('number', blocks.CharBlock(max_length=15)),
+            ('extension', blocks.CharBlock(max_length=4, required=False)),
             ('vanity', blocks.CharBlock(max_length=15, required=False)),
```
While we're touching this file, could we add some `help_text` to better indicate how to use this field?

jimmynotjim replied:
Any thoughts on what that help text should include?

Scotchester commented on cfgov/v1/atomic_elements/molecules.py:
```diff
@@ -257,6 +257,7 @@ class ContactPhone(blocks.StructBlock):
             ('number', blocks.CharBlock(max_length=15)),
+            ('extension', blocks.CharBlock(max_length=4, required=False)),
             ('vanity', blocks.CharBlock(max_length=15, required=False)),
             ('tty', blocks.CharBlock(max_length=15, required=False)),
```
And also put in an explicit label on this field so that it capitalizes the whole thing.

sarahsimpson09 commented:
> Is it safe to assume the TTY extension might not match the primary extension and would need it's own field?

@jimmynotjim yeah, there may likely be times when the only difference between the main line and TTY is the extension, so it would help for TTY to have its own extension field. 

As for the vanity number, since it sits on the same line right next to the main number, I don't think the extension needs to be repeated twice, but open to being wrong if it looks strange without it!

jimmynotjim commented:
@sarahsimpson09 Added the TTY Extension option to the admin

![screen shot 2017-11-07 at 2 11 22 pm](https://user-images.githubusercontent.com/1280430/32515388-8e208c02-c3c5-11e7-9278-af4d985c0b5d.png)

Not sure this is what is supposed to be on the front-end though

![screen shot 2017-11-07 at 2 11 18 pm](https://user-images.githubusercontent.com/1280430/32515402-99377736-c3c5-11e7-85a8-5bc7965af1dd.png)

Do we have any place we've used TTY on the site?

anselmbradford commented on cfgov/jinja2/v1/_includes/molecules/contact-phone.html:
```diff
@@ -37,12 +37,15 @@
         {% endif %}
         {% if phone.tty %}
-        <span>TTY/TDD: {{ format_phone(phone.tty) }}<span>
+        <span>TTY/TDD: {{ format_phone(phone.tty) }}{{ ' Ext ' ~ phone.tty_ext if phone.tty_ext else '' }}<span>
```
No period at the end of `Ext` is stylistically correct @sarahsimpson09 ?

sarahsimpson09 replied:
No, you're right, it should be "Ext."

jimmynotjim commented:
Pushed each number to it's own line and added the `.` to `Ext` text.

![screen shot 2017-11-07 at 5 27 10 pm](https://user-images.githubusercontent.com/1280430/32523477-0e894b3e-c3e1-11e7-89e5-775fd980085f.png)

Also added helper text to the vanity input

![screen shot 2017-11-07 at 5 32 26 pm](https://user-images.githubusercontent.com/1280430/32523602-a3317d74-c3e1-11e7-99f5-b496be645f69.png)

@sarahsimpson09 @anselmbradford want to sign off and I'll merge it?

sarahsimpson09 commented:
@jimmynotjim this looks good to me--thanks!

Scotchester commented:
Pushed a commit to update the migration. Will merge if Travis passes.
